zerver/lib/streams.py

```diff
@@ -658,11 +667,7 @@ def list_to_streams(
         created_streams: List[Stream] = []
     else:
         # autocreate=True path starts here
-        if not user_profile.can_create_streams():
```

Comment:
I see the check on whether a user can or cannot create a stream has been removed here. I believe we need to check an upgraded version of this check here where we say if the stream is public/private and user has permissions to create such a stream we let it through otherwise we error with insufficient permissions?

Reply:
I moved this check up in `create_streams_if_needed` since we need to know if the stream to be created is private or public and check if the user has the permissions to do so.